jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitRunner.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+
+class SamInitRunner {
+    private val samCliExecutable = SamSettings.getInstance().executablePath
+    private var commandLine = GeneralCommandLine(samCliExecutable)
+            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
+            .withParameters("init")
+            .withParameters("--no-input")
+    private lateinit var outputDir: VirtualFile
+    fun applyLocation(location: String) = applyParameter("--location", location)
+
+    fun applyName(name: String) = applyParameter("--name", name)
+
+    fun applyOutputDir(location: VirtualFile) = apply {
+        outputDir = location
+    }
+
+    fun applyRuntime(runtime: Runtime) = applyParameter("--runtime", runtime.toString())
+
+    private fun applyParameter(flag: String, value: String) = apply {
+        commandLine = commandLine.withParameters(flag)
+            .withParameters(value)
+    }
+
+    fun execute() = ApplicationManager.getApplication().runWriteAction {
+        // set output to a temp dir
+        val tempDir = LocalFileSystem.getInstance().findFileByIoFile(createTempDir())
+        applyParameter("--output-dir", tempDir!!.path)
```

Comment:
Let's not use `!!` in mainline code; I prefer the pattern of elvis operatoring (`?:`)  on the potentially null statement and throwing an actual named exception if it is null. So change it to:

```
val tempDir = LocalFileSystem.getInstance().findFileByIoFile(createTempDir()) ?: throw RuntimeException("Cannot create temp file") //No need to put this into message since it's not really a use-facing thing - it shouldn't ever happen.
```

Reply:
done